common/src/auth/ServiceConnectionString.cs

```diff
@@ -0,0 +1,115 @@
+    ///
+    /// This object parses and stores the connection string. It is responsible to provide the authorization token too.
+    /// </remarks>
+    public class ServiceConnectionString
+    {
+        protected static readonly TimeSpan DefaultTokenTimeToLive = TimeSpan.FromMinutes(5);
+
+        /// <summary>
+        /// Creates an instance based on a supplied parser
+        /// </summary>
+        /// <param name="parser">the <see cref="ServiceConnectionStringParser"/> with the connection string content.</param>
+        /// <exception cref="ArgumentNullException">if the provided parser is null.</exception>
+        public ServiceConnectionString(ServiceConnectionStringParser parser)
+        {
+            if (parser == null)
+            {
+                throw new ArgumentNullException(nameof(parser));
+            }
+
+            HostName = parser.HostName;
+            SharedAccessKeyName = parser.SharedAccessKeyName;
+            SharedAccessKey = parser.SharedAccessKey;
+            SharedAccessSignature = parser.SharedAccessSignatureString;
+            ServiceName = parser.ServiceName;
+            HttpsEndpoint = new UriBuilder("https", parser.HostName).Uri;
+        }
+
+        /// <summary>
+        /// Parser for the Service Connection String
+        /// </summary>
+        /// <param name="connectionString"> The Connection String </param>
+        public static ServiceConnectionString Parse(string connectionString)
+        {
+            var parser = ServiceConnectionStringParser.Create(connectionString);
+            return new ServiceConnectionString(parser);
+        }
+
+        /// <summary>
+        /// The Service Name
+        /// </summary>
+        public string ServiceName { get; private set; }
+
+        /// <summary>
+        /// The Service Client Hostname
+        /// </summary>
+        public string HostName { get; private set; }
+
+        /// <summary>
+        /// The Service Client Https Endpoint
+        /// </summary>
+        public Uri HttpsEndpoint { get; private set; }
+
+        /// <summary>
+        /// The Service Audience
+        /// </summary>
+        public string Audience => HostName;
+
+        /// <summary>
+        /// The Service Access Key Name
+        /// </summary>
+        public string SharedAccessKeyName { get; private set; }
+
+        /// <summary>
+        /// The Service Shared Access Key for the specified access policy
+        /// </summary>
+        public string SharedAccessKey { get; private set; }
+
+        /// <summary>
+        /// The Service Shared Access Signature
+        /// </summary>
+        public string SharedAccessSignature { get; private set; }
+
+        /// <summary>
+        /// Returns the shared access signature for authorization
+        /// </summary>
+        public string GetSasToken()
+        {
+            return string.IsNullOrWhiteSpace(SharedAccessSignature) ? BuildToken(out TimeSpan timeToLive) : SharedAccessSignature;
```

Comment:
Readability would be improved IMHO formatted like this:

```
return string.IsNullOrWhiteSpace(SharedAccessSignature)
    ? BuildToken(out TimeSpan timeToList)
    : SharedAccessSignature;
```

Reply:
Sure I have no problem with formatting it.